### PR TITLE
test(crypto): reinforce `polynomial_openings_unittest`

### PR DESCRIPTION
**Pull request by Insun35**
- Add a test polynomial with the same value as another polynomial but with a different address to test if `ShallowRefs` are correctly distinguished.
- Add logic to check the validity of each `poly_openings_vec` of `grouped_poly_openings_vec`.

### CODE REVIEW

**Review by chokobole — APPROVED** [review]
LGTM

File: tachyon/crypto/commitments/polynomial_openings_unittest.cc [hunk 1]
```diff
@@ -118,10 +123,16 @@ TEST_F(PolynomialOpeningsTest, GroupByPolyOracleAndPoints) {
...
   poly_openings.emplace_back(PolyRef(&polys_[2]), PointDeepRef(&points_[2]),
                              polys_[2].Evaluate(points_[2]));
+  poly_openings.emplace_back(PolyRef(&polys_[3]), PointDeepRef(&points_[2]),
+                             polys_[3].Evaluate(points_[2]));
   PolynomialOpeningGrouper<Poly> grouper;
   std::vector<GroupedPolyOraclePair> poly_openings_grouped_by_poly =
       grouper.GroupByPolyOracle(poly_openings);
 
+  // Although |polys_[2]| and |polys_[3]| have the same value, they should be
+  // grouped separately because they are ShallowRef in |PolynomialOpenings|.
```
Comment (TomTaehoonKim) [inline tachyon/crypto/commitments/polynomial_openings_unittest.cc:133]: How about adding `Note(Insun35):` here as well?

**Review by TomTaehoonKim — APPROVED** [review]
Else, LGTM

File: tachyon/crypto/commitments/polynomial_openings_unittest.cc (hunk 1 above)
Comment (chokobole) [reply tachyon/crypto/commitments/polynomial_openings_unittest.cc:133]: NOTE should be all upper cases! `NOTE(Insun35)`

**Review by dongchangYoo — APPROVED** [review]
LGTM

**Review by ashjeong — APPROVED** [review]
LGTM

**Pull request merged**